### PR TITLE
Add a ledgerdb instance for balance/transfer questions

### PR DESCRIPTION
**Pull request by millerdev**
http://manage.dimagi.com/default.asp?158681

@czue @emord

### CODE REVIEW

**Comment by czue** [comment]
nice! thanks!

**Comment by millerdev** [comment]
@czue, @emord is offline and @orangejenny is working on non-vellum stuff this week. Does this seem like something you could review? I'd be happy to do a skype call to walk through it with you if that would help. Alternately we can just wait until one of them has time to review it.

**Comment by czue** [comment]
sure i'll take a pass

**Comment by dannyroberts** [comment]
@millerdev great to see tests!

File: src/commtrack.js [hunk 1]
```diff
@@ -340,6 +347,11 @@ define([
         return mug.parentMug && isInRepeat(mug.parentMug);
     }
 
+    function addLedgerDBInstance(mug, form) {
+        var data = {id: "ledger", src: "jr://instance/ledgerdb"};
```
Comment (czue) [inline src/commtrack.js]: (tiny, but might want to make these constants somewhere)

File: tests/commtrack.js [hunk 2]
```diff
@@ -215,5 +217,63 @@ define([
...
+        it("should not remove ledger instance on delete second Transfer question", function () {
+            util.loadXML("");
+            util.addQuestion("Transfer", "trans");
+            var question = util.addQuestion("Transfer", "question");
+            util.deleteQuestion(question.absolutePath);
+            var xml = util.call("createXML"),
+                $xml = $(xml);
+            assert.equal($xml.find("instance[src='jr://instance/ledgerdb']").length, 1,
+                         "ledger instance should be removed\n" + xml);
+        });
```
Comment (czue) [inline tests/commtrack.js]: these are great

File: src/commtrack.js (hunk 1 above)
Comment (millerdev) [inline src/commtrack.js]: https://github.com/dimagi/Vellum/pull/356

**Pull request merged**